src/OpenTracing/Propagation/BinaryCarrier.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+
+namespace OpenTracing.Propagation
+{
+    /// <summary>
+    /// A <see cref="ITextMapCarrier"/> which allows a byte array to be used as a carrier object.
+    /// </summary>
+    public class BinaryCarrier : IInjectCarrier, IExtractCarrier
```

Comment:
I like them separated. For example in dot net there is a http client in the standard library for  which we need a carrier implementing the inject interface. There are different web servers that need there own carriers just implementing the extract interface. Combining the interfaces would just force non applicable methods throwing NotImplementedExceptions.